pinot-common/src/main/java/com/linkedin/pinot/common/utils/CommonConstants.java

```diff
@@ -95,6 +95,8 @@
         public static final String REALTIME_SEGMENT_FLUSH_TIME = "realtime.segment.flush.threshold.time";
         // Num records threshold in the realtime segment
         public static final String REALTIME_SEGMENT_FLUSH_SIZE = "realtime.segment.flush.threshold.size";
+        // Flush threshold size across all segments
```

Comment:
There's no good heuristic to set this value at this point. The buffers for this are stored in direct memory but the dictionaries are stored on the JVM heap. This means that it's unfortunately very dependent on column cardinality for the on-heap part while the off-heap is unlikely to be a problem.

Both global and per-server names are really bad, I can't think of a good one offhand though. I'll see if I can think of something that makes sense before merging this.